Source/Core/VideoCommon/CommandProcessor.cpp

```diff
@@ -693,16 +695,50 @@ void CommandProcessorManager::HandleUnknownOpcode(u8 cmd_byte, const u8* buffer,
   {
     m_is_fifo_error_seen = true;
 
-    // TODO(Omega): Maybe dump FIFO to file on this error
+    // The panic alert contains an explanatory part that's worded differently depending on the
+    // user's settings, so as to offer the most relevant advice to the user.
+    const char* advice;
+    if (IsOnThread(system) && !system.GetFifo().UseDeterministicGPUThread())
+    {
+      if (!system.GetCoreTiming().UseSyncOnSkipIdle() && !system.GetFifo().UseSyncGPU())
+      {
+// The SyncOnSkipIdle setting is only in the Android GUI, so we use the INI name on other platforms.
+//
+// TODO: Mark the Android string as translatable once we have translations on Android. It's
+// currently untranslatable so translators won't try to look up how they translated "Synchronize
+// GPU Thread" and "On Idle Skipping" and then not find those strings and become confused.
+#ifdef ANDROID
+        advice = "Please change the \"Synchronize GPU Thread\" setting to \"On Idle Skipping\"! "
+                 "It's currently set to \"Never\", which makes this problem very likely to happen.";
+#else
+        // i18n: Please leave SyncOnSkipIdle and True untranslated.
+        // The user needs to enter these terms as-is in an INI file.
+        advice = _trans("Please change the \"SyncOnSkipIdle\" setting to \"True\"! "
+                        "It's currently disabled, which makes this problem very likely to happen.");
+#endif
+      }
+      else
+      {
+        advice = _trans(
+            "This error is usually caused by the emulated GPU desyncing with the emulated CPU. "
+            "Turn off the \"Dual Core\" setting to avoid this.");
+      }
+    }
+    else
+    {
+      advice = _trans(
+          "This error is usually caused by the emulated GPU desyncing with the emulated CPU, "
+          "but your current settings make this unlikely to happen. If this error is stopping the "
+          "game from working, please report it to the developers.");
+    }
+
     PanicAlertFmtT("GFX FIFO: Unknown Opcode ({0:#04x} @ {1}, preprocess={2}).\n"
-                   "This means one of the following:\n"
-                   "* The emulated GPU got desynced, disabling dual core can help\n"
-                   "* Command stream corrupted by some spurious memory bug\n"
-                   "* This really is an unknown opcode (unlikely)\n"
```

Comment:
I must admit, I do have a fondness for this line. I'd like it if you could say something like "This may actually be an unknown opcode" for the case that where the obvious answers don't apply.

Reply:
Haven't we already enumerated and investigated all the possible opcodes by now? In particular, I'm thinking of the last sentence of this comment:

https://github.com/dolphin-emu/dolphin/blob/fbb3db71b2d6df4b4d6c697cfc35b7bdc05c3605/Source/Core/VideoCommon/CommandProcessor.cpp#L644-L654

Reply:
Well, we thought we knew all of the opcodes before too, and well... https://dolphin-emu.org/blog/2022/02/08/dolphin-progress-report-nov-and-dec-2021-jan-2022/#50-15931-disable-unknown-opcode-popup-for-0x1-to-0x7-by-pokechu22

Reply:
Perhaps you could have the message talk about it being an unknown opcode in single core only?

Reply:
Yes, that's what May is suggesting. I would appreciate if someone who knows more about the GC GPU could comment on this, though.

Reply:
FWIW, I think this is an improvement.

As for unknown opcodes... I'm not sure if it's worth showing every user who runs into this (which will be many) a suggestion that both doesn't help them and is also exceedingly unlikely. The current message made sense when Dolphin wasn't mature yet, but I don't think that it's worth keeping that part these days.

Reply:
Maybe poke @Pokechu22 on this; have you ever tried mapping out all the possible opcodes and how the unusual ones behave on hardware?

Reply:
I didn't explicitly test all of them. I determined that *some* invalid opcodes result in hangs, and others are just ignored (though I wasn't scientific in testing if it was a single byte that was ignored or what happened).

For what it's worth, there's still `GX_CMD_UNKNOWN_METRICS` (0x44) and `GX_CMD_INVL_VC` (0x48), neither of which we currently handle. The latter is used by many games and is related to the vertex cache (which is not emulated currently, see #10383); I have no idea what the former does ([this comment](https://github.com/dolphin-emu/dolphin/blob/0b9002ec2a4152607d5f64e879a2c38638a20f14/Source/Core/VideoCommon/OpcodeDecoding.cpp#L206-L213) says it's used by Zelda Four Swords though).

Patent [US6411301](https://patents.google.com/patent/US6411301) (and a few others) has a table that lists various "example" opcodes (Table I, page 30 in the PDF), including xxx for various bits that are presumably ignored (but dolphin doesn't ignore). It also specifies SU_ByPassCmd as `0x60|SUatrr`, where what dolphin refers to as a BPMem command is 0x61 (and that seems to be the only one of that form). Theoretically there could be other hardware that is addressed using command 0x60, 0x62, etc (and I *think* there might be a second patent with a diagram that mentions this, but I can't find it), but I don't think any game actually does this.

It's highly unlikely that any games _intentionally_ use an opcode that triggers our unknown opcode message to do something interesting. There are still cases where games accidentally write something invalid, but there's only been a few of them.